tests/test_operation.py

```diff
@@ -53,29 +61,29 @@ def h_test(cls):
                 return
 
             # not an Expectation
-            # all gaussian ops use the 'A' method
-            self.assertEqual(cls.grad_method, 'A')
-            U = op.heisenberg_tr(2)
+
+            U = op.heisenberg_tr(num_wires=2)
             I = np.eye(*U.shape)
             # first row is always (1,0,0...)
             self.assertAllEqual(U[0,:], I[:,0])
 
             # check the inverse transform
-            V = op.heisenberg_tr(2, inverse=True)
+            V = op.heisenberg_tr(num_wires=2, inverse=True)
             self.assertAlmostEqual(np.linalg.norm(U @ V -I), 0, delta=self.tol)
             self.assertAlmostEqual(np.linalg.norm(V @ U -I), 0, delta=self.tol)
 
-            # compare gradient recipe to numerical gradient
-            h = 1e-7
-            U = op.heisenberg_tr(0)
-            for k in range(cls.num_params):
-                D = op.heisenberg_pd(k)  # using the recipe
-                # using finite difference
-                op.params[k] += h
-                Up = op.heisenberg_tr(0)
-                op.params = par
-                G = (Up-U) / h
-                self.assertAllAlmostEqual(D, G, delta=self.tol)
+            if op.grad_recipe is not None:
+                # compare gradient recipe to numerical gradient
+                h = 1e-7
```

Comment:
another kind of magic number

Reply:
This should get fixed for all tests as per #99